File: the-hurdle-race/the-hurdle-race.go
package thehurdlerace

func hurdleRace(jumpMax int32, heights []int32) int32 {
	heightMax := int32(0)
	for _, h := range heights {
		if h > heightMax {
			heightMax = h
		}
	}

	if heightMax <= jumpMax {
		return 0
	} else {
		return heightMax - jumpMax
	}
}

File: the-hurdle-race/TheHurdleRace.java
import java.util.List;

public class TheHurdleRace {

  public static int hurdleRace(int jumpMax, List<Integer> heights) {
    int heightMax = (int) heights
      .stream()
      .max(Integer::compare)
      .orElseThrow();

    return Math.max(
      0, 
      heightMax - jumpMax
    );
  }

}
